### PR TITLE
adding args to check-ram.rb to allow changing its behavior

### PR DESCRIPTION
**Pull request by sge-babrams**
first pass at rewriting check-ram.rb

The following changes are notable:
- Switched "`free -m`" for vmstat gem to make it more OS agnostic (tested linux & OSX).
- Added flags to use either `--free`, `--used` respectively. This affects whether you are checking resources available or resources in use. The default is to use free to not change the way its working (wouldnt want to make a bunch of people have to change their check defintions).

Here are some typical examples
- default behavior: checks % of RAM free:

```
$ bin/check-ram.rb
CheckRAM OK: 53.46% RAM free
```
- checks % of RAM in use:

```
$ bin/check-ram.rb --used
CheckRAM OK: 48.22% RAM used
```
- checks how many MB of RAM in use:

```
$ bin/check-ram.rb --megabytes --used -w 200 -c 100
CheckRAM CRITICAL: 888 megabytes of RAM used
```

### CODE REVIEW

**Comment by sge-babrams** [comment]
https://github.com/sensu-plugins/sensu-plugins-memory-checks/issues/5

**Comment by sge-babrams** [comment]
@mattyjones @tas50 @maticmeznar had a chance to look at this yet?

**Comment by andyshoe** [comment]
I took a really quick look at this and I think there is an error in your calculation of MB.  The values returned by vmstat are not in KB but rather in number of pages.  The page size may vary from OS to OS, so you need to use the page size data also returned by the vmstat gem to calculate the actual usage.

**Comment by sge-babrams** [comment]
@andyshoe maybe I am misreading something but it returns the number of bytes: https://github.com/threez/ruby-vmstat/blob/master/lib/vmstat/memory.rb though I supose that means that when using --megabytes that calculation is off as its returning Bytes rather than KBs. I will fix that real quick

**Comment by andyshoe** [comment]
From the link you sent:

# @attr [Fixnum] wired

  #   The number of wired pages in the system.
  # @attr [Fixnum] active
  #   The number of active pages in the system.
  # @attr [Fixnum] inactive
  #   The number of inactive pages in the system.

Its the number of pages, like I suggested in my last comment.  Looks like there are also wired_bytes and active_bytes if you wanted to use those instead.  Just as a reminder I'm a Ruby noob so take what I'm saying with a grain of salt, but I'm pretty certain your code is not properly accounting for page size.

**Comment by sge-babrams** [comment]
@andyshoe what I mean is that on each of his functions he times it by page size. Here is an example: https://github.com/threez/ruby-vmstat/blob/master/lib/vmstat/memory.rb#L22

I am also not as strong with the ruby foo as I would like. I am investigating further since it still seems off to me

**Comment by sge-babrams** [comment]
@andyshoe I need to use it like this though I am still showing it a bit off count as far as total ram:

``` ruby
    mem = Vmstat.snapshot.memory
    free_ram = mem.inactive_bytes + mem.free_bytes
    used_ram = mem.wired_bytes + mem.active_bytes
    total_ram = mem.total_bytes
```

**Comment by sge-babrams** [comment]
It looks good to me (regarding the unit sizes) a little off but thats nothing new due to the lack of consensus to use SI prefix or IEC prefix... @mattyjones @tas50 is there a preference in which sensu plugins should use?

**Comment by mattyjones** [comment]
@sge-babrams  I will look at this this week.  I would **really** like to find a way to do this in pure Ruby that also allows for Windows as well.  Absent that I would go with vmstat.

**Comment by sge-babrams** [comment]
@mattyjones I have yet to see a pure ruby implementation that is cross platform, I have been searching for a way and unable to find anything thus settling on vmstat. I hope I am just not searching for the right terms because I agree this is silly that there is no way to do this...From what I have found I think our best bet is OS detection and use vmstat for *nix and some other method for windows maybe a c extension like vmstat is.

**Comment by sge-babrams** [comment]
@mattyjones any word? if we are ready I will rebase and fix merge conflicts

File: Gemfile [hunk 1]
```diff
@@ -2,3 +2,4 @@ source 'https://rubygems.org'
 
 # Specify your gem's dependencies in sensu-plugins-memory-checks.gemspec
 gemspec
+gem 'vmstat'
```
Comment (mattyjones) [inline Gemfile]: @sge-babrams this should be in the gemspec and please hard pin a version

Ex `vmstat = '0.0.1'`

**Comment by mattyjones** [comment]
@sge-babrams  If you update the changelog and fix the dep issue I noted then we are good to merge.

Thanks!

**Comment by sge-babrams** [comment]
@mattyjones updated as requested let me know if there is anything else

**Comment by mattyjones** [comment]
@sge-babrams  will do and I will take care of it but in the futire, you don't need to add a dep to both the gemspec and the gemfile.  The gemfile does nothing but call the gemspec so it is just extra cruft ;p

**Pull request merged**